tests/evergreen/test_api.py

```diff
@@ -181,6 +186,19 @@ def test_versions_by_project_time_window(self, mocked_api, sample_version, mocke
         assert len(windowed_list) == 1
         assert version_list[1]["version_id"] == windowed_list[0].version_id
 
+    def test_most_recent_version_in_project(self, mocked_api, sample_version, mocked_api_response):
+        version_1 = dict(sample_version, **{"version_id": "version_1"})
+        version_2 = dict(sample_version, **{"version_id": "version_2"})
+        version_3 = dict(sample_version, **{"version_id": "version_3"})
+        response_mock = MagicMock()
+        response_mock.status_code = 200
+        response_mock.json.side_effect = [[version_1], [version_2], [version_3]]
+        mocked_api.session.get.return_value = response_mock
+
+        most_recent_version = mocked_api.most_recent_version_in_project("project_id")
+
+        assert most_recent_version.version_id == "version_2"
```

Comment:
I cannot figure out why this is "version_2" instead of "version_1". I do not see how `<api>.session.get()` is getting called before this line runs: https://github.com/evergreen-ci/evergreen.py/pull/49/files#diff-f4c6f3cef9bf46bb4993abeb02e769dbR385

Reply:
The mocking for this looks a little strange. I'm not sure why you are setting the `side_effect` of `json` to a list of lists on line 195. I think just setting the `return_value` to a list of version should get you what you want.
```
response_mock.json.return_value = [version_1, version_2, version_3]
```

Reply:
This was a hacky way of mocking something that could be processed by the pagination code, but I'll remove